narwhal/primary/src/tests/core_tests.rs

```diff
@@ -1028,6 +1083,123 @@ async fn recover_core_expecting_header_of_previous_round() {
     }
 }
 
+#[tokio::test(flavor = "current_thread", start_paused = true)]
+async fn recover_core_latest_headers() {
+    let fixture = CommitteeFixture::builder().randomize_ports(true).build();
+    let committee = fixture.committee();
+    let worker_cache = fixture.shared_worker_cache();
+    let primary = fixture.authorities().last().unwrap();
+    let name = primary.public_key();
+    let other_primary = fixture.authorities().nth(1).unwrap();
+    let signature_service = SignatureService::new(primary.keypair().copy());
+
+    let (_tx_reconfigure, rx_reconfigure) =
+        watch::channel(ReconfigureNotification::NewEpoch(committee.clone()));
+    let (tx_sync_headers, _rx_sync_headers) = test_utils::test_channel!(1);
+    let (tx_certificate_waiter, _rx_certificate_waiter) = test_utils::test_channel!(1);
+    let (_tx_primary_messages, rx_primary_messages) = test_utils::test_channel!(3);
+    let (_tx_headers_loopback, rx_headers_loopback) = test_utils::test_channel!(1);
+    let (_tx_certificates_loopback, rx_certificates_loopback) = test_utils::test_channel!(1);
+    let (_tx_headers, rx_headers) = test_utils::test_channel!(1);
+    let (tx_consensus, _rx_consensus) = test_utils::test_channel!(3);
+    let (tx_parents, _rx_parents) = test_utils::test_channel!(3);
+    let (_tx_consensus_round_updates, rx_consensus_round_updates) = watch::channel(0u64);
+
+    // Create test stores.
+    let (header_store, certificates_store, payload_store) = create_db_stores();
+    let proposer_store = ProposerStore::new_for_tests();
+
+    // add a header to store
+    let builder = types::HeaderBuilder::default();
+    let fixture = CommitteeFixture::builder().randomize_ports(true).build();
+    let primary = fixture.authorities().next().unwrap();
+    let header = builder
+        .author(name.clone())
+        .round(1)
+        .epoch(0)
+        .parents([CertificateDigest::default()].iter().cloned().collect())
+        .with_payload_batch(fixture_batch_with_transactions(10), 0)
+        .build(primary.keypair())
```

Comment:
Should we use the `other_primary` key here since that's the one we'll request the header from?